reconstructSPI/iterative_refinement/expectation_maximization.py

```diff
@@ -363,18 +365,21 @@ def generate_slices(map_3d_f, xy_plane, n_pix, rots):
             of projection of rotated map_3d_f.
             Shape (n_rotations, n_pix, n_pix)
         xyz_rotated : arr
-            Rotated xy plane.
-            Shape (n_pix**2, 3)
+            Rotated xy planes.
+            Shape (n_rotations, 3, n_pix**2)
         """
         n_rotations = rots.shape[0]
-        # map_values interpolation, calculate from map, rots
-        map_3d_f = np.ones_like(map_3d_f)
-        xyz_rotated = np.ones_like(xy_plane)
-
-        size = n_rotations * n_pix**2
-        slices = np.random.normal(size=size)
-        slices = slices.reshape((n_rotations, n_pix, n_pix))
-        return slices, xyz_rotated
+        slices = np.empty((n_rotations, map_3d_f.shape[0], map_3d_f.shape[1]))
+        xy_planes = np.repeat(np.expand_dims(xy_plane, axis=0), n_rotations, axis=0)
+        for i in range(n_rotations):
+            for xy in range(xy_plane.shape[1]):
+                xy_planes[i, :, xy] = rots[i] @ (xy_planes[i, :, xy])
```

Comment:
You just need rotate the same base xy_0 plane array for each rotation. You can allocate an empty that will store the slices, without having to duplicate the xy_0 coordinates with `np.repeat`. See how https://github.com/geoffwoollard/compSPI/blob/stash_simulate/src/simulate.py#L109 is inside the loop and https://github.com/geoffwoollard/compSPI/blob/stash_simulate/src/simulate.py#L104 is defined outside. 

Here things are modularized so that everything is real, so the projection array would not have a complex dtype.